### PR TITLE
Bach/MH: Use setuptools / setup.cfg for all requirements

### PR DESCRIPTION
**Pull request by jansentom**
- [x] Bach: Move deps to setup.cfg
- [x] MH: Move deps to setup.cfg
- [x] Split out BQ reqs and make sure MH always includes those.
- [x] Update READMEs
- [x] Update workflows
- [x] Double check that containers don't use requirements*.txt

To now create your local dev env:
```
python3 -m venv /tmp/venv-deps-test
. /tmp/venv-deps-test/bin/activate'
pip install -e bach[bigquery,dev] -e modelhub[dev]
```

Builds on tightening up reqs: https://github.com/objectiv/objectiv-analytics/pull/1078/files

### CODE REVIEW

**Review by KathiaBarahona — COMMENTED** [review]
just a question. Tho a bit sad there's no way of only restricting the versioning of those packages only for bigquery :/ Got into a rabbit hole investigating how to "overwrite" the requirement based on the command in setup.py, but neh

File: modelhub/setup.cfg [hunk 1]
```diff
@@ -25,7 +25,7 @@ install_requires =
     # sqlalchemy-bigquery (through objectiv-bach) is very specific about the following two deps. List them first to avoid pip getting confused
     sqlalchemy<=1.4.27
     pyarrow<7.0dev
-    objectiv-bach ==0.0.17
+    objectiv-bach[bigquery] ==0.0.17
```
[the remark this answers is not in the document]
Comment (jansentom) [reply modelhub/setup.cfg:28]: We'd like MH to have BQ out of the box, at least for now. That's why it's like this

**Comment by jansentom** [comment]
> just a question. Tho a bit sad there's no way of only restricting the versioning of those packages only for bigquery :/ Got into a rabbit hole investigating how to "overwrite" the requirement based on the command in setup.py, but neh
> 
> The only thing that occurs to me is updating the `install_requires` depending on the markers. https://stackoverflow.com/questions/19672980/can-you-set-conditional-dependencies-for-python-2-and-3-in-setuptools

Yeah, I know about that rabbit hole. We could go that way if we ever want to seperate out BQ support in MH, but I think for now that's not desired (by Vincent/Ivar/Bob)

File: modelhub/setup.cfg (hunk 1 above)
[the remark this answers is not in the document]
Comment (KathiaBarahona) [reply modelhub/setup.cfg:28]: alright!

**Pull request merged**